backend/controller/dal/dal.go

```diff
@@ -490,6 +490,22 @@ func (d *DAL) CreateDeployment(ctx context.Context, language string, moduleSchem
 		}
 	}
 
+	for _, job := range cronJobs {
+		// Start time must be calculated by the caller rather than generated by db
+		// This ensures that nextExecution is after start time, otherwise the job will never be triggered
+		err := tx.CreateCronJob(ctx, sql.CreateCronJobParams{
+			DeploymentKey: deploymentKey,
+			ModuleName:    job.Module,
+			Verb:          job.Verb,
+			StartTime:     job.StartTime,
+			Schedule:      job.Schedule,
+			NextExecution: job.NextExecution,
+		})
+		if err != nil {
+			return model.DeploymentKey{}, fmt.Errorf("%s: %w", "failed to create cron job", translatePGError(err))
```

Comment:
The first string parameter here should just be part of the formatting string:

```suggestion
			return model.DeploymentKey{}, fmt.Errorf("failed to create cron job: %w", translatePGError(err))
```